### PR TITLE
feat: Add  parsing in validate function & Tests  & Documentation

### PR DESCRIPTION
**Pull request by wisnie**
BREAKING CHANGE: The new parsing behaviour changes how `date`, `number` and `string` validators work.

### CODE REVIEW

File: __tests__/property-tests.spec.ts [hunk 1]
```diff
@@ -31,6 +31,10 @@ const throws = <T extends readonly unknown[], Err extends Error>(
   }
 };
 
+const isCoercibleToNum = (value: unknown) => Boolean(Number(value));
```
Comment (mmiszy) [inline __tests__/property-tests.spec.ts:34]: A co z '0'? Lepiej byłoby zrobić `Number.isNaN(`

**Review by mmiszy — CHANGES_REQUESTED** [review]

File: src/parse.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import type { VALIDATORS, ValidatorToType } from './validators';
+import { NUMBER_VALIDATOR, DATE_VALIDATOR, STRING_VALIDATOR } from './validators';
+
+const simplifiedISODateStringRegex = /^[+-]?\d{4}/;
+export const isISODateString = (value: unknown) =>
+  simplifiedISODateStringRegex.test(value as string)
+    ? !Number.isNaN(Number(new Date('1970-01-01T00:00:00.000Z')))
```
Comment (mmiszy) [inline src/parse.ts:7]: Let's add a test for this
```suggestion
    ? !Number.isNaN(Number(new Date(value)))
```

File: src/parse.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  value: unknown,
+): V extends keyof ValidatorToType ? ValidatorToType[V] : unknown => {
+  switch (validator) {
+    case NUMBER_VALIDATOR:
+      if (typeof value === 'string') {
+        if (value.trim() === '') {
+          return value as V extends keyof ValidatorToType ? ValidatorToType[V] : unknown;
+        }
+        const parsedNumber = Number(value);
+        return parsedNumber as V extends keyof ValidatorToType ? ValidatorToType[V] : unknown;
```
Comment (mmiszy) [inline src/parse.ts:20]: Are these type assertions needed?

**Review by mmiszy — CHANGES_REQUESTED** [review]

File: src/parse.ts (hunk 3 above)
Comment (wisnie) [reply src/parse.ts:20]: They can be deleted if we change the return type of this function to ` V extends keyof ValidatorToType ? ValidatorToType[V] : unknown | unknown`.

**Review by mmiszy — APPROVED** [review]

**Pull request merged**